corehq/apps/domain/views/settings.py

```diff
@@ -517,3 +520,87 @@ class ManageDomainMobileWorkersView(ManageMobileWorkersMixin, BaseAdminProjectSe
     page_title = gettext_lazy("Manage Mobile Workers")
     template_name = 'enterprise/manage_mobile_workers.html'
     urlname = 'domain_manage_mobile_workers'
+
+
+@method_decorator(toggles.CUSTOM_DOMAIN_BANNER_ALERTS.required_decorator(), name='dispatch')
+class ManageDomainAlertsView(BaseAdminProjectSettingsView):
+    template_name = 'domain/admin/manage_alerts.html'
+    urlname = 'domain_manage_alerts'
+    page_title = gettext_lazy("Manage Project Alerts")
+
+    @property
+    def page_context(self):
+        return {
+            'form': self.form,
+            'alerts': [
+                {
+                    'active': alert.active,
+                    'html': alert.html,
+                    'id': alert.id,
+                    'created_by_user': alert.created_by_user,
+                }
+                for alert in CommCareHQAlert.objects.filter(created_by_domain=self.domain)
+            ]
+        }
+
+    @cached_property
+    def form(self):
+        if self.request.method == 'POST':
+            return DomainAlertForm(self.request.POST)
+        return DomainAlertForm()
+
+    def post(self, request, *args, **kwargs):
+        command = request.POST.get('command')
+        if command:
+            self._apply_command(request, command)
+        elif self.form.is_valid():
+            self._create_alert()
+            messages.success(request, _("Alert saved!"))
+        else:
+            messages.error(request, _("There was an error saving your alert. Please try again!"))
+            return self.get(request, *args, **kwargs)
+        return HttpResponseRedirect(self.page_url)
+
+    def _apply_command(self, request, command):
+        alert_id = request.POST.get('alert_id')
+        if alert_id:
```

Comment:
Should there be an error if this is false? As is it will silently do nothing, which might be unexpected.

Reply:
Alert ID is passed by JS. It should not be missing.
This is just to avoid errors in case someone messes with the request.

Reply:
I think it's better to fail loudly. From the Zen

```
$ python -m this | grep silently
Errors should never pass silently.
```

Reply:
Not sure I would count this as an error.
I would think here is that, someone wanted to break HQ, so attempts to send the same request but without an alert ID. So we simply not do anything.
If it was possible for user to do this from UI somehow, then I'd consider showing a message that alert ID was missing.

Reply:
Since there is no way to trigger this action in the UI, and it's never expected for a request to be submitted without `alert_id`, then it's better to not add code to handle that scenario.
```suggestion
        assert alert_id
```
It's good to fail early, but please don't silently ignore an unexpected state in code. Silent failures like that make it much more difficult to figure out why things are not working if that condition is triggered for some unforeseen reason.

Reply:
> Silent failures like that make it much more difficult to figure out why things are not working if that condition is triggered for some unforeseen reason.

Good point. Noted.

(Will make the change in code instead of using the commit suggestion here to account for other changes that are needed along)

So, take away here would be to at least fail hard for rare/unexpected scenarios. One could even let the user know though its not worth adding that logic branch in the code. So, for example, here, we had two options
1. raise an exception if alert_id is missing
2. return with an error message "Alert ID not passed"
In terms of UX 2 is better than 1, but code wise that adds an unnecessary "if" condition, so we go with 1 or we should go with 2?

Also, I would think of exceptions as for "unhandled" scenarios. So if an exception is expected, it should be accounted (rescued) for and handled gracefully. Hence, I am in split about using an assertion here.

Reply:
> take away here would be to at least fail hard for rare/unexpected scenarios

Yes, exactly. In this case you have said

> It should not be missing. This is just to avoid errors in case someone messes with the request.

To me that would indicate it was a rare/unexpected scenario, and it is probably not worth adding special error handling for that, so go with option 1. If you later notice, while monitoring Sentry after rolling out the change, that this unexpected condition is triggered then it would be worth it to figure out why it is happening and fix the issue at that time.

Reply:
https://github.com/dimagi/commcare-hq/pull/33592/commits/7feb32bb00bd9d78db8613cfa658e8650aab8e6e